### ISSUE TITLE
Filter Reset button doesn't work

### ISSUE THREAD

**Issue by happywillow0** [tachiyomiorg/tachiyomi#8015]
### Steps to reproduce

1. Open the filter in any source. 
2. Change any of the options
3. Hit the reset button to reset filter options

### Expected behavior

The selected filter options should revert back to the default / first options. 

### Actual behavior

Any selected filters remain selected. 
Interestingly, the actual results page behind the filter select tray in the gets refreshed. 

### Tachiyomi version

Preview r4904

### Android version

Android 13

### Device

Pixel 6a

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.6](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by geospiza-fortis** [comment]
I'm interested in taking this, I've reproduced this on the tip in master in an emulator. I can't reproduce on my device with the last stable version 0.13.6, so it must be some kind of regression.

**Comment by ghostbear** [comment]
0.13.6 isn't remotely close to the current code base, better check with preview version

**Comment by geospiza-fortis** [comment]
Alright, I ran through git bisect from the v0.13.6 tag, and found two places where the filter button behavior changes. I tested by going to an extension (MangaDex), pressing on the filter menu, clicking on the first available checkbox (Has available chapter), and then hitting reset.

d4b764fa317ddcb04b8fefb982cd6b9a6dfc1598 is the first bad commit where the filter button acts strangely. It will reset the checkbox once, but subsequent resets will do nothing. 

I then did a bisect between that and master. The first bad commit where the reset button is completely broken is in 6b2b21edfa63f01b956ec27386b0c729be2e6d18. In addition to the broken filter behavior, I notice that the filter menu moves to the top of the menu.
